tasks/upload.yml

```diff
@@ -22,7 +22,7 @@
     interface: "{{ os_images_interface | default(omit, true) }}"
     name: "{{ item.name ~ '-kernel' }}"
     state: present
-    is_public: "{{ item.is_public | default(os_images_public) | bool }}"
+    is_public: "{{ item.is_public | item.visibility == 'public' | default(os_images_public) | bool }}"
```

Comment:
@markgoddard I think this should make it backwards compatible? That way people can use either variable

Reply:
Almost.

```
is_public: "{{ item.is_public | default(item.visibility == 'public') | default(os_images_public) | bool }}"
```

But this is still using the old is_public argument rather than visibility, so you're not really getting the new functionality.

Do we also need to provide a global `os_images_visibility` (with a default based on os_images_public)?